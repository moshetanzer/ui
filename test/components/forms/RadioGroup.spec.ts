import { describe, it, expect } from 'vitest'
import { URadioGroup } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('RadioGroup', () => {
  it.each([
    ['basic case', {}],
    ['with legend', { props: { legend: 'Radio Group Legend' } }],
    ['with slot legend', { slots: { legend: () => 'Radio Group Slot Legend' } }],
    ['with options and legend', { props: { options: [{ label: 'Option 1', value: 'option1' }, { label: 'Option 2', value: 'option2' }] } }],
    ['with disabled', { props: { disabled: true, options: [{ label: 'Option 1', value: 'option1' }, { label: 'Option 2', value: 'option2' }] } }],
    ['with custom color', { props: { color: 'red', options: [{ label: 'Option 1', value: 'option1' }, { label: 'Option 2', value: 'option2' }] } }],
    ['with label', { props: { label: 'label' } }]
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof URadioGroup.props>) => {
    const html = await ComponentRender(nameOrHtml, options, URadioGroup)
    expect(html).toMatchSnapshot()
  })
})
