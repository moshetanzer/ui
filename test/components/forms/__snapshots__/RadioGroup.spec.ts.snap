// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RadioGroup > renders basic case correctly 1`] = `
"<div class="relative flex items-start">
  <fieldset>
    <!--v-if-->
  </fieldset>
</div>"
`;

exports[`RadioGroup > renders with custom color correctly 1`] = `
"<div class="relative flex items-start">
  <fieldset>
    <!--v-if-->
    <div class="relative flex items-start">
      <div class="flex items-center h-5"><input id="4" type="radio" class="h-4 w-4 dark:checked:bg-current dark:checked:border-transparent disabled:opacity-50 disabled:cursor-not-allowed focus:ring-0 focus:ring-transparent focus:ring-offset-transparent form-radio form-radio bg-white dark:bg-gray-900 border border-gray-300 dark:border-gray-700 focus-visible:ring-2 focus-visible:ring-red-500 dark:focus-visible:ring-red-400 focus-visible:ring-offset-2 focus-visible:ring-offset-white dark:focus-visible:ring-offset-gray-900 text-red-500 dark:text-red-400" value="option1"></div>
      <div class="ms-3 flex flex-col"><label for="4" class="text-sm font-medium text-gray-700 dark:text-gray-200">Option 1
          <!--v-if-->
        </label>
        <!--v-if-->
      </div>
    </div>
    <div class="relative flex items-start">
      <div class="flex items-center h-5"><input id="5" type="radio" class="h-4 w-4 dark:checked:bg-current dark:checked:border-transparent disabled:opacity-50 disabled:cursor-not-allowed focus:ring-0 focus:ring-transparent focus:ring-offset-transparent form-radio form-radio bg-white dark:bg-gray-900 border border-gray-300 dark:border-gray-700 focus-visible:ring-2 focus-visible:ring-red-500 dark:focus-visible:ring-red-400 focus-visible:ring-offset-2 focus-visible:ring-offset-white dark:focus-visible:ring-offset-gray-900 text-red-500 dark:text-red-400" value="option2"></div>
      <div class="ms-3 flex flex-col"><label for="5" class="text-sm font-medium text-gray-700 dark:text-gray-200">Option 2
          <!--v-if-->
        </label>
        <!--v-if-->
      </div>
    </div>
  </fieldset>
</div>"
`;

exports[`RadioGroup > renders with disabled correctly 1`] = `
"<div class="relative flex items-start">
  <fieldset>
    <!--v-if-->
    <div class="relative flex items-start">
      <div class="flex items-center h-5"><input id="2" disabled="" type="radio" class="h-4 w-4 dark:checked:bg-current dark:checked:border-transparent disabled:opacity-50 disabled:cursor-not-allowed focus:ring-0 focus:ring-transparent focus:ring-offset-transparent form-radio form-radio bg-white dark:bg-gray-900 border border-gray-300 dark:border-gray-700 focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 focus-visible:ring-offset-2 focus-visible:ring-offset-white dark:focus-visible:ring-offset-gray-900 text-primary-500 dark:text-primary-400" value="option1"></div>
      <div class="ms-3 flex flex-col"><label for="2" class="text-sm font-medium text-gray-700 dark:text-gray-200">Option 1
          <!--v-if-->
        </label>
        <!--v-if-->
      </div>
    </div>
    <div class="relative flex items-start">
      <div class="flex items-center h-5"><input id="3" disabled="" type="radio" class="h-4 w-4 dark:checked:bg-current dark:checked:border-transparent disabled:opacity-50 disabled:cursor-not-allowed focus:ring-0 focus:ring-transparent focus:ring-offset-transparent form-radio form-radio bg-white dark:bg-gray-900 border border-gray-300 dark:border-gray-700 focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 focus-visible:ring-offset-2 focus-visible:ring-offset-white dark:focus-visible:ring-offset-gray-900 text-primary-500 dark:text-primary-400" value="option2"></div>
      <div class="ms-3 flex flex-col"><label for="3" class="text-sm font-medium text-gray-700 dark:text-gray-200">Option 2
          <!--v-if-->
        </label>
        <!--v-if-->
      </div>
    </div>
  </fieldset>
</div>"
`;

exports[`RadioGroup > renders with label correctly 1`] = `
"<div class="relative flex items-start">
  <fieldset label="label">
    <!--v-if-->
  </fieldset>
</div>"
`;

exports[`RadioGroup > renders with legend correctly 1`] = `
"<div class="relative flex items-start">
  <fieldset>
    <legend class="text-sm font-medium text-gray-700 dark:text-gray-200 mb-1">Radio Group Legend</legend>
  </fieldset>
</div>"
`;

exports[`RadioGroup > renders with options and legend correctly 1`] = `
"<div class="relative flex items-start">
  <fieldset>
    <!--v-if-->
    <div class="relative flex items-start">
      <div class="flex items-center h-5"><input id="0" type="radio" class="h-4 w-4 dark:checked:bg-current dark:checked:border-transparent disabled:opacity-50 disabled:cursor-not-allowed focus:ring-0 focus:ring-transparent focus:ring-offset-transparent form-radio form-radio bg-white dark:bg-gray-900 border border-gray-300 dark:border-gray-700 focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 focus-visible:ring-offset-2 focus-visible:ring-offset-white dark:focus-visible:ring-offset-gray-900 text-primary-500 dark:text-primary-400" value="option1"></div>
      <div class="ms-3 flex flex-col"><label for="0" class="text-sm font-medium text-gray-700 dark:text-gray-200">Option 1
          <!--v-if-->
        </label>
        <!--v-if-->
      </div>
    </div>
    <div class="relative flex items-start">
      <div class="flex items-center h-5"><input id="1" type="radio" class="h-4 w-4 dark:checked:bg-current dark:checked:border-transparent disabled:opacity-50 disabled:cursor-not-allowed focus:ring-0 focus:ring-transparent focus:ring-offset-transparent form-radio form-radio bg-white dark:bg-gray-900 border border-gray-300 dark:border-gray-700 focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 focus-visible:ring-offset-2 focus-visible:ring-offset-white dark:focus-visible:ring-offset-gray-900 text-primary-500 dark:text-primary-400" value="option2"></div>
      <div class="ms-3 flex flex-col"><label for="1" class="text-sm font-medium text-gray-700 dark:text-gray-200">Option 2
          <!--v-if-->
        </label>
        <!--v-if-->
      </div>
    </div>
  </fieldset>
</div>"
`;

exports[`RadioGroup > renders with slot legend correctly 1`] = `
"<div class="relative flex items-start">
  <fieldset>
    <legend class="text-sm font-medium text-gray-700 dark:text-gray-200 mb-1">Radio Group Slot Legend</legend>
  </fieldset>
</div>"
`;
